### PR TITLE
fix(selector): should have image select option when no Dockerfile found

### LINKED ISSUE #1656: Copilot doesn't prompt for an existing image name if workspace is empty
### Description

My current workspace doesn't have a Dockerfile, so when I run `copilot init` or `copilot svc init` I expected the CLI to ask me if I'd like to enter an existing image name. But the CLI instead prompts me for:
```
What is the path to the Dockerfile for my-svc? [? for help] 
```

This seems to be happening because of the `WorkspaceSelector` where if no Dockerfile is found we prompt them for a path:
https://github.com/aws/copilot-cli/blob/037cae99ec055f7c875d0efa23563cb39d198ae9/internal/pkg/term/selector/selector.go#L529-L542

Instead, we should be displaying the following prompt:
```
Which Dockerfile would you like to use for my-svc?
> Enter custom path for your Dockerfile
> Use an existing image instead
```

### PR DESCRIPTION
**Pull request by iamhopaul123**
fixes #1656.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/term/selector/selector.go [hunk 1]
```diff
@@ -510,27 +512,23 @@ func (s *WorkspaceSelect) retrieveWorkspaceWorkloads() ([]string, error) {
...
+		prompt.WithFinalMessage("Dockerfile:"),
+	)
+	if err != nil {
+		return "", fmt.Errorf("select Dockerfile: %w", err)
 	}
-
-	var notExistErr *workspace.ErrDockerfileNotFound
-	if !errors.As(err, &notExistErr) {
-		return "", err
+	if sel != dockerfilePromptUseCustom {
```
Comment (bvtujo) [inline internal/pkg/term/selector/selector.go:534]: slick

File: internal/pkg/term/selector/selector_test.go [hunk 2]
```diff
@@ -1226,23 +1233,23 @@ func TestWorkspaceSelect_Dockerfile(t *testing.T) {
...
-					}),
+					gomock.Eq(dockerfileOptions),
 					gomock.Any(),
 				).Return("frontend/Dockerfile", nil)
 			},
 			wantedErr:        nil,
 			wantedDockerfile: "frontend/Dockerfile",
 		},
-		"prompts user for custom path if fail to find Dockerfiles": {
+		"prompts user for custom path": {
```
Comment (bvtujo) [inline internal/pkg/term/selector/selector_test.go:1243]: Can we make sure there's a test for the case where no dockerfiles are found and we only ask "Input custom dockerfile" and "Select existing image?"

Comment (iamhopaul123) [reply internal/pkg/term/selector/selector_test.go:1243]: :+1:

Comment (iamhopaul123) [reply internal/pkg/term/selector/selector_test.go:1243]: done!

File: internal/pkg/workspace/workspace.go [hunk 3]
```diff
@@ -468,11 +468,6 @@ func (ws *Workspace) ListDockerfiles() ([]string, error) {
 			}
 		}
 	}
-	if len(directories) == 0 {
```
Comment (efekarakus) [inline internal/pkg/workspace/workspace.go:471]: nice i like this change of returning just an empty list instead of an error if no dockerfiles are found

**Review by efekarakus — APPROVED** [review]

**Review by bvtujo — APPROVED** [review]
Nice, thank you so much for this!

**Pull request merged**